internal/pkg/deploy/deploy.go

```diff
@@ -16,3 +36,135 @@ type ResourceEvent struct {
 	Status       string
 	StatusReason string
 }
+
+type resourceGetter interface {
+	GetResourcesByTags(resourceType string, tags map[string]string) ([]string, error)
+}
+
+type configStoreClient interface {
+	GetEnvironment(appName string, environmentName string) (*config.Environment, error)
+	ListEnvironments(appName string) ([]*config.Environment, error)
+	GetService(appName, svcName string) (*config.Service, error)
+}
+
+// Store is in charge of fetching the service deployment information.
+type Store struct {
+	rgClient     resourceGetter
+	configStore  configStoreClient
+	initRgClient func(*Store, string, string) error
+}
+
+// NewStore returns a new store.
+func NewStore() (*Store, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("connect to config store: %w", err)
+	}
+	return &Store{
+		configStore: store,
+		initRgClient: func(s *Store, appName, envName string) error {
+			env, err := s.configStore.GetEnvironment(appName, envName)
+			if err != nil {
+				return fmt.Errorf("get environment %s, %w", envName, err)
+			}
+			p := session.NewProvider()
+			sess, err := p.FromRole(env.ManagerRoleARN, env.Region)
+			if err != nil {
+				return fmt.Errorf("create new session from env role: %w", err)
+			}
+			s.rgClient = rg.New(sess)
+			return nil
+		},
+	}, nil
+}
+
+// ListDeployedServices returns all the deployed service in a specific environment of an application.
+func (s *Store) ListDeployedServices(appName string, envName string) ([]*config.Service, error) {
```

Comment:
I think most packages should only return structs defined by themselves.  

In this situation, the external API of `deploy` becomes tightly coupled with `config`. Instead, I think we can just return identifiers and let callers decide whether they need additional information. For example:
```suggestion
// ListDeployedServices returns the names of deployed services in an environment part of an application.
func (s *Store) ListDeployedServices(appName string, envName string) ([]string, error) {
```

This way clients have no idea whether we use `config` under the hood or not, and we get flexibility if we ever want to change the internals.

Reply:
> I think the rule-of-thumb that I follow is most packages should only return structs defined by themselves.

Agree with the rule, which makes a lot of sense to me!